packages/core/src/components/steps-modal/steps-modal.tsx

```diff
@@ -0,0 +1,100 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  Host,
+  Prop,
+  forceUpdate,
+  h,
+} from '@stencil/core'
+
+@Component({
+  tag: 'atom-steps-modal',
+  styleUrl: 'steps-modal.scss',
+  shadow: false,
+})
+export class AtomStepsModal {
+  @Prop() steps: number = 0
+  @Prop({ mutable: true }) currentStep: number = 0
+  @Prop() trigger?: string
+  @Prop() stepsTitles: string
+  @Prop({ mutable: true }) isOpen: boolean = false
+
+  @Event() atomFinish: EventEmitter
+  @Event() atomCancel: EventEmitter
+  @Event() atomNextStep: EventEmitter
+  @Event() atomPreviousStep: EventEmitter
+  @Event() atomCloseClick: EventEmitter
+  @Event() atomDidDismiss: EventEmitter
+  @Event() atomDidPresent: EventEmitter
+
+  @Element() el!: HTMLElement
+
+  private stepsTitlesArray: string[] = []
+
+  componentWillLoad() {
+    this.stepsTitlesArray = this.stepsTitles.split(',')
+  }
+
+  private handleStep = (step: number) => {
+    this.currentStep = step
+  }
+
+  private handlePrimaryClick = () => {
+    if (this.currentStep === this.steps - 1) {
```

Comment:
question: does make sense for `currentStep` to assume 1 as the default value instead?

Reply:
this difference is impacting the progress bar
![image](https://github.com/user-attachments/assets/20581098-9328-4fb3-9f8a-2b81efa59ca7)